docs/two-phase-commit-transactions.md

```diff
@@ -0,0 +1,175 @@
+
+## Configuration
+
+The configuration for Two-phase Commit Transactions is the same as the one for the transaction API.
+
+For example, you can set the following configuration when you use Cassandra:
+```
+# Comma separated contact points
+scalar.db.contact_points=cassandra
+
+# Port number for all the contact points. Default port number for each database is used if empty.
+scalar.db.contact_port=9042
+
+# Credential information to access the database
+scalar.db.username=cassandra
+scalar.db.password=cassandra
+
+# Storage implementation. Either cassandra or cosmos or dynamo or jdbc can be set. Default storage is cassandra.
+scalar.db.storage=cassandra
+```
+
+Please see [Getting Started](getting-started.md) for configurations of other databases/storages.
+
+### Scalar DB server
+
+You can also execute Two-phase Commit Transactions through the Scalar DB server.
+You don't need a special configuration for Two-phase Commit Transactions, so you can follow [the Scalar DB server document](scalardb-server.md) to use it.
+
+## How to execute Two-phase Commit Transactions
+
+This section explains how to execute Two-phase Commit Transactions.
+
+Like a well-known two-phase commit protocol, there are two roles, a coordinator and a participant, that collaboratively execute a single transaction.
+The coordinator process first starts a transaction, and the participant processes join the transaction after that.
+
+### Get a TwoPhaseCommitTransactionManager instance
+
+First, you need to get a `TwoPhaseCommitTransactionManager` instance to execute Two-phase Commit Transactions.
+You can use `TransactionFactory` to get a `TwoPhaseCommitTransactionManager` instance as follows:
+```Java
+TransactionFactory factory = new TransactionFactory(new DatabaseConfig(new File("<configuration file path>")));
+TwoPhaseCommitTransactionManager manager = factory.getTwoPhaseCommitTransactionManager();
+```
+
+### Start a transaction
+
+You can start a transaction as follows:
+```Java
+TwoPhaseCommitTransaction tx = manager.start();
+```
+
+The process/application that starts the transaction acts as a coordinator, as mentioned.
+
+You can also start a transaction by specifying a transaction ID as follows:
+```Java
+TwoPhaseCommitTransaction tx = manager.start("<transaction ID>");
+```
+
+And, you can get the transaction ID with `getId()` as follows:
+```Java
+tx.getId();
+```
+
+### Join the transaction
+
+You can join the transaction that has been started by the coordinator as follows:
+```Java
+TwoPhaseCommitTransaction tx = manager.join("<transaction ID>")
+```
+
+You need to specify the transaction ID associated with the transaction that the coordinator has started.
+
+The process/application that joins the transaction acts as a participant, as mentioned.
+
+#### Resume the transaction
+
+You can get the transaction object (the `TwoPhaseCommitTransaction` instance) that you have already joined with `TwoPhaseCommitTransactionManager.resume()`:
```

Comment:
Sorry, I think I'm not fully sure, but a coordinator process also calls resume? 
Or, it is only called by a participant process?

Reply:
It is only called by a participant process. A coordinator can't resume its transaction in the current implementation.